### DISCUSSION TITLE
cgroup hierarchy doesn't follow kubernetes recommended best practices

### DISCUSSION THREAD

**Discussion by cyrus-mc** [bottlerocket-os/bottlerocket#1679] [General]
https://github.com/kubernetes/community/blob/master/contributors/design-proposals/node/node-allocatable.md#recommended-cgroups-setup

As highlighted above there should be a system.slice/cgroup and a podruntime.slice/cgroup that correspond to systemReserved and kubeReserved settings. Looking at the setup of Bottlerocket is appears there is just system.slice under which kubelet and runtime plus all system components run.

Was there a reason for this design decision? As it doesn't map nicely to the configuration settings k8s provides.

**Comment by webern** [comment]
Thanks for bringing this to our attention, and for the pointer to the proposal. Your link also led us to find this https://kubernetes.io/docs/tasks/administer-cluster/reserve-compute-resources/#kube-reserved which suggests "runtime.slice" and "system.slice" and points to the design proposal. We'll review our current slice setup and make sure it's aligned with kubelet functionality, unless it's something you'd like to contribute.

**Reply by webern** [reply]
P.S. I created issue #1681 for this. Thanks again!
